### ISSUE TITLE
Checked and indeterminate should be independent Checkbox states

### ISSUE THREAD

**Issue by joemaffei** [mantinedev/mantine#2573]
### What package has an issue

@mantine/core

### Describe the bug

The Checkbox component's [implementation](https://github.com/mantinedev/mantine/blob/master/src/mantine-core/src/Checkbox/Checkbox.tsx#L116) sets the `checked` attribute of the input based first on the value of `indeterminate`, then the value of `checked`. That makes impossible to set the input to indeterminate and unchecked.

When setting the `indeterminate` prop to true, the input's indeterminate property should be set (source: [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox#indeterminate_state_checkboxes)) so that screen readers can announce the correct state. This is not 100% covered by screen readers out there (source: [GitHub comment](https://github.com/mui/material-ui/issues/20476#issuecomment-611468844)) but it might be an improvement over the current implementation.

Interestingly, when I inspected the docs page for the Checkbox component, if I store the indeterminate checkbox input as a global variable (temp1) and set `temp1.indeterminate` to true, the background color disappears!

### What version of @mantine/hooks page do you have in package.json?

5.4.1

### If possible, please include a link to a codesandbox with the reproduced problem

https://codesandbox.io/s/hardcore-lamport-0608vx?file=/src/App.tsx

### Do you know how to fix the issue

Yes

### Are you willing to participate in fixing this issue and create a pull request with the fix

Yes

**Comment by joemaffei** [comment]
Another possible solution would be to use ARIA attributes on a focusable/clickable div: https://www.w3.org/WAI/ARIA/apg/example-index/checkbox/checkbox-mixed.html

**Comment by joemaffei** [comment]
There is a link to a Codesandbox in the GitHub link in the description. I just forked it and adapted it to Mantine, which helps to see the difference in styling between native indeterminate checkboxes and Mantine's implementation: https://5yiehh.csb.app/

**Comment by rtivital** [comment]
Fixed in 5.5.0

**Comment by joemaffei** [comment]
@rtivital Is the indeterminate property of the HTML input set according to the prop? I imagine you would need to do something like this:
```js
useEffect(() => {
  ref.indeterminate = indeterminate;
}, [indeterminate]);
```
Another option would be to set `aria-checked="mixed"` when checked and indeterminate but, as I understand it, that's only appropriate on non-input elements with `role="checkbox"`.
